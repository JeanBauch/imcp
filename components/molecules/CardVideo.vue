<template>
	<div class="flex flex-col justify-center">
		<h3 class="text-3xl text-black md:text-[40px] mb-6 mt-10 md:my-10 font-semibold">
			Escola de música <span :class="themeColors.textColor">em ação</span>
		</h3>
		<img
			class="rounded-3xl max-h-[189px] md:max-h-[800px]"
			src="https://images.unsplash.com/photo-1465847899084-d164df4dedc6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80"
			alt=""
			srcset=""
		>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';
export default Vue.extend({
	data () {
		return {
			themeColors: {
				textColor: `text-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-500`,
			},
			show: false,
			teste: '',
		};
	},
	props: {
		color: {
			required: false,
			type: String,
			default: 'orange',
			validator: (value: string) => [
				'yellow',
				'orange',
				'bordo',
				'beige',
				'blue',
			].includes(value.toLowerCase()),
		},
	},
});
</script>
