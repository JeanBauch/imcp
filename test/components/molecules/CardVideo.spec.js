import { shallowMount } from '@vue/test-utils';

import CardVideo from '~/components/molecules/CardVideo.vue';

describe('Components:molecules:CardVideo', () => {
	const factory = (values = {}) => {
		return shallowMount(CardVideo, {
			propsData: {
				...values,
			},
		});
	};

	describe('Snapshots', () => {
		it('should render component with correctly props', () => {
			const wrapper = factory({ color: 'orange' });

			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render correctly data as with props datas', () => {
			const wrapper = factory({ color: 'orange' });

			expect(wrapper.find('.text-primary-orange-500').exists()).toBeTruthy();
		});

		it('should render correctly data with color:orange', async () => {
			const wrapper = factory({ color: 'orange' });

			await wrapper.setData({ themeColors: { textColor: 'text-primary-orange-500' } });

			expect(wrapper.vm.themeColors.textColor).toBe('text-primary-orange-500');
		});

		it('should render correctly data with color:blue', async () => {
			const wrapper = factory({ color: 'blue' });

			await wrapper.setData({ themeColors: { textColor: 'text-secondary-blue-500' } });

			expect(wrapper.vm.themeColors.textColor).toBe('text-secondary-blue-500');
		});
	});
});
