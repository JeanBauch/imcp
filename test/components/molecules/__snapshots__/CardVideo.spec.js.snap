// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:molecules:CardVideo Snapshots should render component with correctly props 1`] = `
"<div class=\\"flex flex-col justify-center\\">
  <h3 class=\\"text-3xl text-black md:text-[40px] mb-6 mt-10 md:my-10 font-semibold\\">
    Escola de música <span class=\\"text-primary-orange-500\\">em ação</span></h3> <img src=\\"https://images.unsplash.com/photo-1465847899084-d164df4dedc6?ixlib=rb-1.2.1&amp;ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&amp;auto=format&amp;fit=crop&amp;w=1170&amp;q=80\\" alt=\\"\\" srcset=\\"\\" class=\\"rounded-3xl max-h-[189px] md:max-h-[800px]\\">
</div>"
`;
